Updated Date and Time Functions List (#3908)
opensearch-project/documentation-website: _search-plugins/sql/functions.md

tan | `tan(number T) -> double` | `SELECT tan(0.5) FROM my-index LIMIT 1`

## Date and time

> Functions marked with * have future PPL implementation on roadmap.

Reviewer: ```suggestion
Functions marked with * are only available in SQL. 
```

Reviewer 2: Addressed in https://github.com/opensearch-project/documentation-website/pull/3908/commits/e033e7fa5264764d0226feee54fb4d99acc9afbd
